services/datamanager/data_manager.go

```diff
@@ -50,6 +50,8 @@ type Service interface {
 	resource.Resource
 	// Sync will sync data stored on the machine to the cloud.
 	Sync(ctx context.Context, extra map[string]interface{}) error
+	UploadImageToDataset(ctx context.Context, image []byte, datasetIDs, tags []string,
```

Comment:
don't we need the mime type?
can we also have an image.Image version even if just for the go sdk?

Reply:
yes! we definitely should have the MIME type. Lemme add that. 

Do you want the `image.Image` version instead or in addition? If we pass in MIME type then it’s not an issue.